src/universe/UniverseInferenceValidator.java

```diff
@@ -0,0 +1,97 @@
+package universe;
+
+import checkers.inference.InferenceValidator;
+import checkers.inference.InferenceVisitor;
+import com.sun.source.tree.ParameterizedTypeTree;
+import com.sun.source.tree.Tree;
+import org.checkerframework.common.basetype.BaseTypeChecker;
+import org.checkerframework.framework.type.AnnotatedTypeFactory;
+import org.checkerframework.framework.type.AnnotatedTypeMirror;
+import org.checkerframework.framework.type.AnnotatedTypeParameterBounds;
+import org.checkerframework.javacutil.TreeUtils;
+
+import javax.lang.model.element.TypeElement;
+import java.util.List;
+
+import static universe.UniverseInferenceChecker.BOTTOM;
+import static universe.UniverseInferenceChecker.LOST;
+import static universe.UniverseInferenceChecker.REP;
+
+/**
+ * This type validator ensures correct usage of ownership modifiers or generates
+ * constraints to ensure well-formedness.
+ */
+public class UniverseInferenceValidator extends InferenceValidator {
```

Comment:
Can you compare this to `UniverseValidator` and make sure they are consistent?

Reply:
Yes, they are consistent